include/ginkgo/core/base/range.hpp

```diff
@@ -101,6 +101,17 @@ struct span {
      */
     constexpr bool is_valid() const { return begin < end; }
 
+    /**
+     * Returns the length of a span.
+     *
+     * @return `this->end - this->begin`
+     */
+    constexpr size_type length() const
+    {
+        GKO_ASSERT(is_valid());
```

Comment:
`static_assert`?

Reply:
I dont think that is possible as begin and end need to be accessed, which are not always constexpr

Reply:
I guess we use it in a dynamic context as well even if this is constexpr. Thanks for checking.

Reply:
Yes, in our code we use runtime defined versions in many places. But in some user's code, if the span in defined at compile time, the constexpr would kick in, I think.